### PR TITLE
Event - SQS SEP/Reference Server implementation and refactor

### PR DESCRIPTION
**Pull request by stfung77**
-SQS implementation for both SEP and Reference Server
- both servers can be configured to use either Kafka or SQS
- refactoring of event handling and publishing
- refactor Sep server config file to match reference server
- use ScheduleExecutorService in the SQS listener to poll multiple queues

### CODE REVIEW

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java [hunk 1]
```diff
@@ -24,12 +24,31 @@ public class AnchorEventProcessor {
 
   public void handleQuoteEvent(QuoteEvent event) {
     Log.debug(String.format("Received quote event: %s", event));
+    switch (event.getType()) {
+      case "quote_created":
+        break;
+      default:
+        Log.debug("error: anchor_platform_event - invalid message type '%s'%n", event.getType());
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java:31]: I think `Log.error` is better.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java [hunk 2]
```diff
@@ -24,12 +24,31 @@ public class AnchorEventProcessor {
...
+    }
   }
 
   public void handleTransactionEvent(TransactionEvent event) {
     Log.debug(String.format("Received transaction event: %s", event));
+    switch (event.getType()) {
+      case "transaction_created":
+      case "transaction_payment_received":
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java:39]: There is a modification https://github.com/stellar/stellar-anchor-platform/blob/main/design/eventSchema/event.yml#L54 

We do not have "transaction_payment_received" event type any more.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java [hunk 3]
```diff
@@ -24,12 +24,31 @@ public class AnchorEventProcessor {
...
+        handleTransactionPaymentReceivedEvent(event);
+      case "transaction_status_changed":
+      case "transaction_error":
+      default:
+        Log.debug("error: anchor_platform_event - invalid message type '%s'%n", event.getType());
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java:44]: `Log.error`?

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  public SqsListener(SqsListenerSettings sqsListenerSettings, AnchorEventProcessor processor) {
+    this.processor = processor;
+    this.sqsClient =
+        AmazonSQSAsyncClientBuilder.standard()
+            .withRegion(sqsListenerSettings.getRegion())
+            .withCredentials(
+                new AWSStaticCredentialsProvider(
+                    new BasicAWSCredentials(
+                        sqsListenerSettings.getAccessKey(), sqsListenerSettings.getSecretKey())))
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java:30]: There are multiple ways that AWS credentials can be configured. We probably need to re-factor all the secret configurations after this PR is merged.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.concurrent.*;
+import javax.annotation.PreDestroy;
+import org.stellar.anchor.event.models.QuoteEvent;
+import org.stellar.anchor.event.models.TransactionEvent;
+import org.stellar.anchor.reference.config.SqsListenerSettings;
+import org.stellar.anchor.util.Log;
+
+public class SqsListener extends AbstractEventListener {
+  private final AnchorEventProcessor processor;
+  private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(5);
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java:19]: Can we make this part of the configuration and default to `5`?

Comment (lijamie98) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java:19]: There are several calls of `scheduleAtFixRate`. We also need to validate the thread pool size is greater than the number of calls to maximize the concurrency.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+  class SqsListen implements Runnable {
+    private final String queue;
+
+    public SqsListen(String queue) {
+      this.queue = queue;
+    }
+
+    public void run() {
+      Log.debug("SQS event consumer started for queue: " + queue);
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java:63]: `Log.info`?

We will need to define the logging strategy for `debug`, `info`, `warning` and `error`.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+      Gson gson = new Gson();
+      try {
+        String queueUrl = sqsClient.getQueueUrl(queue).getQueueUrl();
+        List<Message> messages = getSqsMessages(queue);
+        while (messages.size() > 0) {
+
+          for (Message message : messages) {
+            String eventClass =
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/SqsListener.java:71]: nit: Is this `eventType` or `eventClass`?

**Review by lijamie98 — APPROVED** [review]
LGTM. 

Left a couple of comments.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java (hunk 2 above)
Comment (stfung77) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/AnchorEventProcessor.java:39]: will address the schema change in a separate PR

**Pull request merged**